service/api_logic/teams_logic.py

```diff
@@ -1,34 +1,60 @@
 import json
 from database.azure_blob_storage.save_get_blob import get_all_blob_indexes_from_db, get_blob_data_for_all_sports
+from service.api_logic.scripts import get_sport_by_name
+from api.routes.dto import UniversalResponseDTO
+from exept.handle_exeptions import handle_exceptions
+from database.models import TeamIndex, Sport
+from flask import Response
+
 
 TEAMS_JSON = "teams.json"
 TEAM_KEY = "team"
 DATA_KEY = "data"
 SPORT_KEY = "sport"
 RESPONSE_KEY = "response"
 
+
+@handle_exceptions
 def get_teams(session):
-    blob_indexes = get_all_blob_indexes_from_db(session, TEAMS_JSON)
-    result = get_blob_data_for_all_sports(session, blob_indexes)
-    data = json.loads(result)
-    filtered_data = []
+        teams_blob_indexes = get_all_blob_indexes_from_db(session, TEAMS_JSON)
+        sport_result = get_blob_data_for_all_sports(session, teams_blob_indexes)
+        sport_result_json = json.loads(sport_result)
+        filtered_data = [
+            processed_data
+            for teams_data in sport_result_json
+            if (processed_data := process_blob_data(teams_data))
+        ]
+        return filtered_data, 200
```

Comment:
Status code again. Remove.

By the way, if you see multiple places with same return statement (in your case it would be: "return something, status_code") it's a good sign you need a model/class/DTO created.

Follow DRY principle (and SOLID, of course)